xbmc/network/WebServer.cpp

```diff
@@ -1035,10 +1035,43 @@ void CWebServer::ContentReaderFreeCallback(void *cls)
 #endif
 }
 
+// local helper
+static void MHDPanicHandler(void* unused, const char* file, unsigned int line, const char *reason)
+{
+  CLog::Log(LOGSEVERE, "MHD serious error: reason \"%s\" in file \"%s\" at line %ui", reason ? reason : "",
+            file ? file : "", line);
+  throw exception("MHD serious error"); // FIXME: better solution?
```

Comment:
Panic function is called in case of serious internal error. You can't shutdown webserver only as internal lists may be corrupted. You even can't kill webserver as you don't know which threads are generated by webserver.
It should not be a problem as chances that it will be ever called are really small.